docs/enhancements/sep-tarball-client.md

```diff
@@ -0,0 +1,170 @@
+    QueryInfos []*sonobuoy.QueryInfo
+}
+
+// Plugin is a single plugin that executed during a Sonobuoy run.
+type Plugin struct {
+    // Name is the name of the plugin.
+    Name string
+    // Results is a map of plugin result filenames to the file contents.
+    Results map[string][]byte
+}
+
+// PodLog are all the logs gathered from an individual pod.
+type PodLog struct {
+    // Name is the name of the pod.
+    Name string
+    // Logs are the logs in io.Reader form because they can be quite large.
+    Logs io.Reader
+}
+```
+
+Some sample code using this interface:
+
+```go
+data, _ := ioutil.ReadFile("sonobuoy.tar.gz")
+tarballClient := sonobuoy.NewTarball(data)
+
+// Getting resources (namespaced or not) should be really familiar.
+_ := tarballClient.Resources().Core().Nodes()
+_ := tarballClient.Resources().Core().Pods("default")
+
+// Some information about the hosts configuration.
+hosts := tarballClient.Hosts()
+for _, host := range hosts {
+    fmt.Println("is the kubelet on", host.Name, "containerized?", host.KubeletConfig["containerized"])
+    if host.StatusCode!= "200" {
+        fmt.Println(host.Name, "is not ok according to the healthz endpoint")
+    }
+}
+
+// Metadata about the Sonobuoy run itself.
+metadata := tarballClient.Metadata()
+fmt.Println(metadata.Config()) // a sonobuoy.Config
+for _, queryInfo := range metadata.QueryTime() {
+    fmt.Println(queryInfo) // a sonobuoy.queryData (TODO: Export.)
+}
+
+// Getting plugin files should be pretty generic to support all plugins.
+// Current assumptions include that a plugin will have some output.
+plugins := tarballClient.Plugins()
+for _, plugin := range plugins {
+    fmt.Println("Found the following results for", plugin.Name)
+    for filename, contents := range plugin.Results {
+        fmt.Println(filename)
+        fmt.Println(string(result))
+    }
+}
+
+podLogs := tarballClient.PodLogs()
+for _, podLog := range podLogs {
+    fmt.Println("pod name", podLog.Name)
+    out, _ := ioutil.ReadAll(podLog.Logs)
+    fmt.Println(string(out))
+}
+
+// This is a *version.Info from kubernetes.
+serverVersion := tarballClient.ServerVersion()
+fmt.Println(serverVersion)
+
+```
+
+### User Stories
+
+#### https://scanner.heptio.com
+
+- Scanner provides a user interface to Sonobuoy Results. It has to read through
+tarballs generated by Sonobuoy and extract files and support multiple tarball
+versions. It should not have to implement this custom logic and should be able
+to leverage this API.
+
+#### Custom plugin user
+
+- A user has developed a custom plugin and wants an easy way to get the files
+the tarball contains without too much hassle.
+
+#### Tarballs from various Sonobuy versions
+
+- A user has a collection of tarballs across multiple versions of Sonobuoy and
+wants to read the data without having to worry about the implementation of the
+tarball schema directly.
+
+## Unresolved Questions
+
+* None at this time.
```

Comment:
That's interesting. I think that could be a neat modification to sonobuoy itself. It could provide a checksum at write time, but it's very hard to know if a run is complete. We'd have to be pretty flexible on that.

Like, if resources don't exist on the cluster they won't exist in the tarball. Did we collect them or did the run fail on collecting a specific resource? 

Still, I think it's possible. It will require some modifications in sonobuoy core (at the very least to write a checksum to the archive) as well as thought into how exactly to know what a complete sonobuoy run looks like.